org.eclipse.xtext.ui.shared/src/org/eclipse/xtext/ui/shared/internal/Activator.java

```diff
@@ -72,14 +83,89 @@ protected void initializeInjector(BundleContext context) {
+	 * @throws CoreException Will be thrown in any case
+	 * @since 2.13
+	 */
+	protected void handleGuiceInitializationError(Throwable t) throws CoreException {
+		Bundle bundleXtextRuntime = Platform.getBundle("org.eclipse.xtext");
+		Bundle bundleXtextIde = Platform.getBundle("org.eclipse.xtext.ide");
+		Bundle bundleXtextUi = Platform.getBundle("org.eclipse.xtext.ui");
+		Bundle bundleXtextUiShared = Platform.getBundle(PLUGIN_ID);
+
+		StringBuilder msg = new StringBuilder();
+		msg.append("Xtext configuration error!\n");
+		msg.append("This error might indicate missing version constraints of installed Xtext based bundles.\n");
+		
+		msg.append("The following bundles were detected that require Xtext UI:\n");
+		
+		getBundlesRequiringXtextUi()
+			.forEach((bundle) -> msg.append("- "+getBundleInfo(bundle) +"\n"));
+		
+		msg.append("The installed Xtext bundle versions are:\n");
+		Lists.newArrayList(bundleXtextRuntime, bundleXtextIde, bundleXtextUi, bundleXtextUiShared)
+		.forEach((bundle) -> msg.append("- "+getBundleInfo(bundle) + "\n"));
+
+		if (bundleXtextRuntime.getVersion().compareTo(bundleXtextUiShared.getVersion()) > 0) {
+			msg.append("Runtime bundle is NEWER than UI bundles! Please make sure that both bundles are installed with the same version!\n");
+		}
+		throw new CoreException(new Status(IStatus.ERROR, getBundle().getSymbolicName(), msg.toString(), t));
+	}
+	
+	private String getBundleInfo (Bundle bundle) {
+		String state = null;
+		switch (bundle.getState()) {
+			case Bundle.UNINSTALLED: state = "UNINSTALLED"; break;
+			case Bundle.INSTALLED: state = "INSTALLED"; break;
+			case Bundle.RESOLVED: state = "RESOLVED"; break;
+			case Bundle.STARTING: state = "STARTING"; break;
+			case Bundle.STOPPING: state = "STOPPING"; break;
+			case Bundle.ACTIVE: state = "ACTIVE"; break;
+		}
+		return bundle.getSymbolicName() + " " + bundle.getVersion() + " ["+state+"]";
+	}
+
+	/**
+	 * Retrieves all bundles that declare a RequireBundle dependency on org.eclipse.xtext.ui, excluding bundles from Xtext itself.
+	 * @since 2.13
+	 */
+	public List<Bundle> getBundlesRequiringXtextUi() {
+		return Arrays.asList(getBundle().getBundleContext().getBundles()).stream()
+			.filter((bundle)-> // exclude Xtext's own bundles
+				!Arrays.stream(new String[] {"org.eclipse.xtext","org.eclipse.xtend"}).anyMatch((s)->bundle.getSymbolicName().startsWith(s))
+			)
+			.filter(
+				(bundle) -> {
+				try {
+					ManifestElement[] requireBundle = ManifestElement.parseHeader(Constants.REQUIRE_BUNDLE, bundle.getHeaders().get(Constants.REQUIRE_BUNDLE));
+					if (requireBundle == null)
+						return false;
+					Stream<ManifestElement> requireElements = Arrays.stream(requireBundle);
+					return requireElements
+						.map((e)->e.getValueComponents())
+						.flatMap((e)->Arrays.stream(e))
+						.anyMatch((bundleName) -> Platform.getBundle("org.eclipse.xtext.ui").getSymbolicName().equals(bundleName))
```

Comment:
Platform.getBundle("org.eclipse.xtext.ui").getSymbolicName() is executed for all elements in the stream until the first match is found. Maybe use a local variable instead? I know it doesn't really matter since this is only the error handling code and not executed in a tight loop.